caddyhttp/httpserver/server.go

```diff
@@ -160,6 +160,18 @@ func (s *Server) Serve(ln net.Listener) error {
 		ln = tcpKeepAliveListener{TCPListener: tcpLn}
 	}
 
+	if ProxyProtocolSubnets != "" {
+		// We decode proxy protocol after the TCP wrapper, but before
+		// anything else. As long as the wrapper happens before things
+		// that care about RemoteAddr, we should be ok.
+		fln, err := newFilteredPROXYListener(ln, ProxyProtocolTimeout, ProxyProtocolSubnets)
+		if err != nil {
+			fmt.Printf("[ERROR] configuring Proxy Protocol support: %v\n", err)
```

Comment:
I think this error should be a show-stopper (why would you want it to continue without proxy protocol support -- I assume if you enable it it's because your service depends on it), and Serve() must not return errors before serving begins. Move this wrapping of the listener up to the Listen() method instead, and make the error return. (If you have good reasoning to simply make this a warning, let me know.)